main.go

```diff
@@ -197,6 +203,10 @@ func main() {
 			if err != nil {
 				fmt.Printf("Could not download the Firehose events file:\n%v\n", err)
 				errorEvents.With(prometheus.Labels{"type": "s3-get"}).Inc()
+				if strings.Contains(string(err.Error()), "ExpiredToken") {
+					// update session
+					s3client = client(region)
```

Comment:
This is to support having Vault Agent running as a sidecar, so when the STS token expires, the client is updated and we can continue the loop.